rest/src/main/java/org/n52/io/extension/RenderingHintsExtension.java

```diff
@@ -74,7 +74,11 @@ public void addExtraMetadataFieldNames(DatasetOutput output) {
     }
 
     private boolean hasRenderingHints(DatasetOutput output) {
-        return hasSeriesConfiguration(output) || hasPhenomenonConfiguration(output);
+        return hasDatasetParameters(output) && (hasSeriesConfiguration(output) || hasPhenomenonConfiguration(output));
+    }
+
+    private boolean hasDatasetParameters(DatasetOutput output) {
+        return output.getDatasetParameters() != null;
```

Comment:
Yes querying with Parameters `?fields=extras` is not going to yield the exspected results with this implementation.

Regarding Solution 1:
- This could possibly create a big overhead when Extensions build up their own stack to get the Data directly from the DAO when its not provided by the AbstractOutput. I dislike this option.

Regarding Solution 2:
A System with serializationFlags would indeed solve the problem. I imagine something like:

- serializationFlags completely replace the proposed System of using `setValue()` to control the Elements in the AbstractOutput. They are set on Object Construction. All Field-Data would be added to the AbstractOutput, but getter-Methods would only return data if flag is set. Extensions could choose to ignore/overwrite flags to retrieve Data regardless. 

- Flags are implemented as: "requested by extension"-flag and are set on Construction. `setValue()` respects those flags while setting Data. getter-Methods return null if flag is set(to prevent renderng to normal output). Problematic: Extensions are computed after Output was created -> flags need to be known beforehand.